### DISCUSSION TITLE
Beta testesr for SPAN with Solar Battery.

### DISCUSSION THREAD

**Discussion by pavandave** [cayossarian/span#14] [General]
Hello,

I wrote an enhancement so that HA can read battery percentages from SPAN. I have a Tesla Powerwall 3 and the HA integration for that is broken so I wrote this to help me monitor and automate based off of battery levels.

Looking for some people to test this out . -- I've run it in my lab and prod environment with no issues. Bill also ran this and without a battery he was getting a value of 95. Would be curious what other peoples SPANs are returning.

Current beta - https://github.com/SpanPanel/Span/tree/0.0.13-beta

Any feedback would be greatly appreciated. 

Thanks,

Screenshots:
The new sensor.
![361995431-7f7f250b-bc43-49bd-a6bc-a72d7997a825](https://github.com/user-attachments/assets/aed1abda-8b19-4e26-9b73-93840435c391)
Comparison of data between HA and another dashboard.
![362063943-3eedf308-9594-4536-8db8-27b17b49ee38](https://github.com/user-attachments/assets/e20ad671-fc4e-4bba-976c-081f4fe8e7bd)

**Comment by dwsminywi** [comment]
I might be interested in testing it if it will work with a SolarEdge system.  The SPAN app shows the battery  and solar production, so if the solar production could also be added in that would be helpful.   I http link goes to a 404 page for me, so maybe this is abandoned?

**Reply by cayossarian** [reply]
The 1.0.3 version is out, thank you.  We will have another beta soon but for another issue regarding entity names to support renaming.
